### PR TITLE
Remove deprecated methods in google-cloud-core

### PR DESCRIPTION
**Pull request by shinfan**

### CODE REVIEW

File: google-cloud-core/src/main/java/com/google/cloud/HttpServiceOptions.java [hunk 1]
```diff
@@ -203,7 +199,6 @@ public int getConnectTimeout() {
     return connectTimeout;
   }
```
Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/HttpServiceOptions.java]: You missed `connectTimeout`

Comment (shinfan) [inline google-cloud-core/src/main/java/com/google/cloud/HttpServiceOptions.java]: Done

**Comment by shinfan** [comment]
PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM after at least one test passes.

**Comment by shinfan** [comment]
AppVeyor build succeeded. Going to merge.

**Pull request merged**